Assets/MixedRealityToolkit/Inspectors/ExtensionTemplates/ExtensionServiceCreator.cs

```diff
@@ -401,6 +442,12 @@ public async Task BeginAssetCreationProcess()
             if (Result == CreateResult.Error)
                 return;
 
+            await Task.Delay(100);
```

Comment:
Some of the Task.Delays below make sense to me (i.e. waiting for compile) but these here I don't fully understand - CreateTextAssetFromTemplate appears to just be doing string replacement and WriteTextAssetToDisk LOOKS to be the sync flavor - Also it looks like each section is doing an unrelated thing (i.e. one for inspector, one for template, etc.)

What's the wait for (and could it be commented?). If it not necessary, can we remove it?

Usually anytime you see random 10-1000ms awaits in code it's a red flag.

Reply:
Those waits serve no purpose other than to provide visual flow as assets are being created.

Reply:
Can you add a comment about this?